File: elixir/gigasecond/lib/gigasecond.ex
defmodule Gigasecond do
  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) ::
          {{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}
  def from({{year, month, day}, {hours, minutes, seconds}}) do
    date = Date.new!(year, month, day)
    time = Time.new!(hours, minutes, seconds)

    DateTime.new!(date, time)
    |> DateTime.add(1_000_000_000)
    |> convert()
  end

  defp convert(%DateTime{
         year: year,
         month: month,
         day: day,
         hour: hours,
         minute: minutes,
         second: seconds
       }) do
    {{year, month, day}, {hours, minutes, seconds}}
  end
end

File: csharp/gigasecond/Gigasecond.cs
public static class Gigasecond
{
    /// <summary>
    /// One giga is equivalent to one billion
    /// </summary>
    private static readonly int GIGA = 1_000_000_000;
    /// <summary>
    /// Calculates the exact moment when it will be one gigasecond after the
    /// given date.
    /// </summary>
    /// <param name="moment">
    /// A moment chosen to calculate the gigaversary.
    /// </param>
    /// <returns>Gigaversary of the given moment</returns>
    public static DateTime Add(DateTime moment)
        => moment.AddSeconds(GIGA);
}